src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -687,15 +687,13 @@ public void RegisterMessageHandler(Func<Message, CancellationToken, Task> handle
             ReceivingAmqpLink receivingAmqpLink = await this.ReceiveLinkManager.GetOrCreateAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);
             Source source = (Source)receivingAmqpLink.Settings.Source;
             string tempSessionId;
-            if (!source.FilterSet.TryGetValue(AmqpClientConstants.SessionFilterName, out tempSessionId))
+            if (!source.FilterSet.TryGetValue(AmqpClientConstants.SessionFilterName, out tempSessionId) || string.IsNullOrWhiteSpace(tempSessionId))
```

Comment:
Doesn't need a new event. Its caller (`AcceptMessageSessionAsync`) logs all the exceptions it receives